DOCS-2998: TC 2.1 update alert manage article (#1212)
truenas/documentation: content/TrueCommand/Alerts/AlertManage.md

  - /truecommand/tcgettingstarted/configurealert/
---

TrueCommand alerts provide visual warnings for monitored systems that require attention.
Alerts are either generated by the monitored system or an alert rule created in TrueCommand.

TrueCommand provides three alert screen options:
* **All Alerts** which displays all system alerts discovered by TrueCommand on the systems it monitors.
* **Alert Rules** which allows adminstrators and users with permissions to configure alerts for monitored systems.
* **Alert Services** which allows adminstrators to configure communication plugins.

> Alerts inidications display on the **All Alerts**, **Dashboard** and **Systems** screens. 

Reviewer: typo
```suggestion
Alerts indications display on the **All Alerts**, **Dashboard** and **Systems** screens. 
```

{{< tabs "Alert Screens" >}}
{{< tab "All Alerts" >}}

## Viewing Alerts

To see all alerts discovered by TrueCommand, open the **Configure** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> menu and click **All Alerts**.
Administrator accounts can see all system alerts.
A non-administrator account can only view alerts according to their team and user account permissions.

![AllAlerts](/images/TrueCommand/2.1/AllAlerts.png "All Alerts")

### Viewing Alerts by System

Alerts generated by a monitored system display on both the **Systems** screen and the system card or detail single system view on the **Dashboard** as a number to the right of the system name.

![DashboardSystemAlerts](/images/TrueCommand/2.0/DashboardSystemAlerts.png "Dashboard System Alerts")

To view all alerts for a single system, from the **Dashboard** and click on a system name with alerts indicated. That system's details screen displays with the alerts listed in the **Alerts** area of the screen.

![DashboardSystemDetailAlerts](/images/TrueCommand/2.1/DashboardSystemDetailAlerts.png "Dashboard System Detail Alerts")

> Click the <mat-icon _ngcontent-bbg-c138="" role="img" fontset="mdi" fonticon="mdi-comment-text-multiple" class="mat-icon notranslate mdi mdi-comment-text-multiple mat-icon-no-color" aria-hidden="true" data-mat-icon-type="font" data-mat-icon-name="mdi-comment-text-multiple" data-mat-icon-namespace="mdi"></mat-icon> icon in the **Alerts** area of the system detail screen to display the **Notice Details** side panel with information about the alert and any user comments made.

Reviewer: This <mat-icon> code should be able to be pared down to the <span class="iconify" data-icon="mdi:comment-text-multiple"></span>
```suggestion
Click the <span class="iconify" data-icon="mdi:mdi-comment-text-multiple"></span> icon in the **Alerts** area of the system detail screen to display the **Notice Details** side panel with information about the alert and any user comments made.
```

To view all alerts from the **Systems** screen system list, click the **Configure** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> menu and then click **Systems**.

![SystemsList](/images/TrueCommand/2.1/SystemsList.png "Systems List")

> Just as with the systems detail alerts, click on the <mat-icon _ngcontent-bbg-c138="" role="img" fontset="mdi" fonticon="mdi-comment-text-multiple" class="mat-icon notranslate mdi mdi-comment-text-multiple mat-icon-no-color" aria-hidden="true" data-mat-icon-type="font" data-mat-icon-name="mdi-comment-text-multiple" data-mat-icon-namespace="mdi"></mat-icon> icon in the **COMMENTS** column to display the **Notice Details** side panel with information about the alert and any user comments made.

Reviewer: As above, pared down icon call:
```suggestion
Just as with the systems detail alerts, click on the <span class="iconify" data-icon="mdi:comment-text-multiple"></span> icon in the **COMMENTS** column to display the **Notice Details** side panel with information about the alert and any user comments made.
```

## Resolving Alerts

You can move an alert in the **All Alerts** screen to the **RESOLVED** column by clicking the **Resolve** <i class="material-icons" aria-hidden="true" title="Check">check</i> to the right of the delete icon.

To resolve multiple alerts, select each alert checkbox and then click **RESOLVE SELECTED**.

> Click the <mat-icon _ngcontent-bbg-c138="" role="img" fontset="mdi" fonticon="mdi-comment-text-multiple" class="mat-icon notranslate mdi mdi-comment-text-multiple mat-icon-no-color" aria-hidden="true" data-mat-icon-type="font" data-mat-icon-name="mdi-comment-text-multiple" data-mat-icon-namespace="mdi"></mat-icon> icon in the **COMMENTS** column to display the **Notice Details** side panel with information about the alert and any user comments made.

Reviewer: ```suggestion
Click the <span class="iconify" data-icon="mdi:comment-text-multiple"></span> icon in the **COMMENTS** column to display the **Notice Details** side panel with information about the alert and any user comments made.
```

![AllAlertDetails](/images/TrueCommand/2.1/AllAlertDetails.png "All Alert Details")

To resolve an alert on the system detail screen **Alerts** area, click on the **Resolve alert notice** <i class="material-icons" aria-hidden="true" title="Check">check</i> to the left of the comments icon.

![SystemResolveAlertNotice](/images/TrueCommand/2.1/SystemResolveAlertNotice.png "System Resolve Alert Notice")

## Deleting Alerts

> Administrator accounts can delete an alert by clicking the delete icon <i class="material-icons" aria-hidden="true" title="Delete">delete</i>.

Reviewer: Made consistent with previous usage.
```suggestion
Administrator accounts can delete an alert by clicking the <i class="material-icons" aria-hidden="true" title="Delete">delete</i> icon.
```

3. Finished: To create the new alert rule, click **CREATE ALERT**. To start over, click **RESET**.

{{< /tab >}}
{{< tab "Alert Services">}}

{{< hint info >}}
Configurable alert services are only available for local installations or containerized TrueCommand deployments.
TrueCommand Cloud instances are preconfigured to use email alerts; PagerDuty is not an option.
{{< /hint >}}

TrueCommand uses different service plugins to expand how alerts are communicated to individual users or administrators.
Individual user accounts can use these services to manage how that account is notified of a system alert.  

## Configuring Alert Services

To configure an alert service plugin, open the **Configure** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> menu and click **Alert Services**. There are two services listed:

* **PagerDuty** is a plugin to configure a pager to receive an alert.
* **SMTP Email** is a plugin to configure system and user email services.

![Alert Services Screen](/images/TrueCommand/2.1/AlertServiceScreen.png "Alert Services")

Each plugin has three options:

> * **Send test email** <mat-icon _ngcontent-ath-c575="" role="img" fontset="mdi" fonticon="mdi-test-tube" class="mat-icon notranslate mdi mdi-test-tube mat-icon-no-color" aria-hidden="true" ng-reflect-font-set="mdi" ng-reflect-font-icon="mdi-test-tube"></mat-icon> 

Reviewer: ```suggestion
* **Send test email** <span class="iconify" data-icon="mdi:test-tube"></span>
```

* **Configure plugin** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> 

> * **Clear plugin configuration** <mat-icon _ngcontent-ath-c200="" role="img" class="mat-icon notranslate material-icons mat-icon-no-color" aria-hidden="true">remove_circle</mat-icon> 

Reviewer: Is this one working on builds?
```suggestion
* **Clear plugin configuration** <mat-icon _ngcontent-ath-c200="" role="img" class="mat-icon notranslate material-icons mat-icon-no-color" aria-hidden="true">remove_circle</mat-icon> 
```

Author: This was the original icon link content that worked I just moved it to display after the 'label' for it for consistent parallel construction in the bullet list. 
My local build worked, but the hint box I added after it was messed up so I removed the hint box and it worked without issues.

## Configuring SMTP Email
 
{{< hint info >}}
Before proceeding, verify that the sending mailserver has TLS enabled.  
TrueCommand cannot send emails through a mailserver without TLS.
An email address must exist on the users profile page to recieve emails.
{{< /hint >}}

To properly configure SMTP email:

1. Enter values in all fields on the **SMTP Email** screen and then click **SAVE**:
   
   ![AlertServicesSMTPEmailptions](/images/TrueCommand/2.0/AlertServicesSMTPEmailptions.png "Alert Services: SMTP Email options")

   * **Mailserver** for example *smtp.gmail.com*
   * **Mailserver port** number
   * **Auth user** email address for plain authentication, for example, *adminuser@yourmail.com*
   * **Auth pass** password for the plain authentication; for a *No-Auth* SMTP configuration, leave the password field blank

>    * **From** is the what sends the eamil (i.e., no-reply@TrueCommand.io),or allows you to customize the sender field of the email 

Reviewer: typos
```suggestion
   * **From** is what sends the email (i.e., no-reply@TrueCommand.io), or allows you to customize the sender field of the email 
```

>   Click **Test** on the **SMTP Email** configuration screen to verify that the configuarion is correct.

Reviewer: typo
```suggestion
  Click **Test** on the **SMTP Email** configuration screen to verify that the configuration is correct.
```

  If you did not receive a test alert email, check the values entered for accuracy.

2. Click on the avatar to the right of the **Configure** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> menu and then click **Profile** from the dropdown list.
     
   ![EditUserProfile](/images/TrueCommand/2.1/EditUserProfile.png "Edit User Profile")

3. Type the email for that user in the **Email** field and click **SAVE CHANGES**.
   The first time you set up SMTP email a **VERIFY EMAIL** button displays below the **Email** field to the left of the **Enable 2FA** checkbox.

   The system should automatically send a test email to the specified email address but if not click **VERIFY EMAIL**.  
   A **Confirmation** dialog box displays with a field where you type or copy/paste the code sent in the email. Doing this verifies the email.
   
   ![SMTPSetupVerifyEmail](/images/TrueCommand/2.1/SMTPSetupVerifyEmail.png "SMTP Setup Verify Email")

## Configuring PagerDuty

> Open the **Configure Plugin** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> for PagerDuty. In the
> **Authtoken** section, enter your **Service Integration Key from PagerDuty**.  If you have an active subscription with PagerDuty, this key should be available to you. 

Reviewer: Kind of an odd line break. Simple suggestion to break on the sentence.
```suggestion
Open the **Configure Plugin** <i class="material-icons" aria-hidden="true" title="Settings">settings</i> for PagerDuty.
In the **Authtoken** section, enter your **Service Integration Key from PagerDuty**. If you have an active subscription with PagerDuty, this key should be available to you. 
```

Author: That was an oops... I didn't catch it. Thanks
